### PR TITLE
fix: usage of deprecated `Node.js 16 action`

### LINKED ISSUE #2968: `rustfmt` uses deprecated `Node.js 16 actions`

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #2968.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -264,20 +263,7 @@ jobs:
...
-          default: true
-          profile: minimal
-          components: rustfmt
-
-      - name: Check formatting
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+      - run: cargo fmt --all -- --check
```
[the remark this answers is not in the document]
Comment (hamirmahal) [reply .github/workflows/ci.yaml:266]: I'm testing it out, and it looks like `rustfmt` can use `rustc 1.77.1 (7cf61ebde 2024-03-27)` while `validate-deps` uses `rustc 1.77.2 (25ef9e3d8 2024-04-09)`.

[the remark this answers is not in the document]
Comment (hamirmahal) [reply .github/workflows/ci.yaml:266]: It looks like `rustfmt` can use `rustc 1.77.1 (7cf61ebde 2024-03-27)` while `validate-deps` uses `rustc 1.77.2 (25ef9e3d8 2024-04-09)`.

[the remark this answers is not in the document]
Comment (lpil) [reply .github/workflows/ci.yaml:266]: We'll need to ensure that they all use the most recent stable.

[the remark this answers is not in the document]
Comment (hamirmahal) [reply .github/workflows/ci.yaml:266]: Okay, I'll look into this.

[the remark this answers is not in the document]
Comment (hamirmahal) [reply .github/workflows/ci.yaml:275]: Okay, with the latest changes, steps should use consistent versions of Rust.

[the remark this answers is not in the document]
Comment (lpil) [reply .github/workflows/ci.yaml:275]: Sorry, I'm being a bit slow to understand. Don't we need to remove all the instances of this action? There still are some. Also don't we need to install Rust to make sure it's the most recent version? Thank you

[the remark this answers is not in the document]
Comment (hamirmahal) [reply .github/workflows/ci.yaml:275]:
```yml
      - name: Install Gleam
        uses: actions-rs/cargo@v1
        with:
          command: install
          args: "--path compiler-cli --target ${{ matrix.target }} --debug --locked"
          use-cross: ${{ matrix.use-cross }}
        if: ${{ matrix.run-integration-tests }}

      - name: Run tests
        uses: actions-rs/cargo@v1
        with:
          command: test
          args: "--workspace --target ${{ matrix.target }}"
          use-cross: ${{ matrix.use-cross }}
```

is a bit more complex to replace; this pull request just gets rid of `uses: actions-rs/cargo@v1` where it is easy to do so.

[the remark this answers is not in the document]
Comment (hamirmahal) [reply .github/workflows/ci.yaml:275]: > Also don't we need to install Rust to make sure it's the most recent version? Thank you

You're welcome! This action uses the following operating systems.

```yml
          - os: ubuntu-latest
          - os: macos-12 # intel
          - os: macos-14 # aarch64
          - os: windows-latest
```

[The Action runner image for each of these](https://github.com/actions/runner-images/tree/main/images) comes with Rust.

https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#rust-tools
https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#rust-tools
https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md#rust-tools
https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#rust-tools

[the remark this answers is not in the document]
Comment (lpil) [reply .github/workflows/ci.yaml:275]: This doesn't say that it uses the most recent versions of Rust, it says it uses a specific version. Do they get upgraded immediately? We need to be sure we are using latest stable.

[the remark this answers is not in the document]
Comment (hamirmahal) [reply .github/workflows/ci.yaml:275]: This pull request leaves the

```yml
      - name: Install Rust toolchain
        uses: actions-rs/toolchain@v1
        with:
          toolchain: stable
          override: true
          default: true
          profile: minimal
          components: clippy
```

parts of `.github/workflows/ci.yaml` unchanged.

That should handle using the latest stable version of Rust, no?

[the remark this answers is not in the document]
Comment (lpil) [reply .github/workflows/ci.yaml:275]: Sounds good!

**Review by lpil — APPROVED** [review]
Thank you

**Comment by hamirmahal** [comment]
You're welcome!

File: .github/workflows/ci.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hamirmahal) [reply .github/workflows/ci.yaml:275]: You're welcome!

**Pull request merged**